restapi.go

```diff
@@ -2339,3 +2361,215 @@ func (s *Session) RelationshipsMutualGet(userID string) (mf []*User, err error)
 	err = unmarshal(body, &mf)
 	return
 }
+
+// ------------------------------------------------------------------------------------------------
+// Functions specific to application (slash) commands
+// ------------------------------------------------------------------------------------------------
+
+// ApplicationCommandCreate creates an global application command and returns it.
+// appID       : The application ID. If empty - s.State.User.ID. Note: works only with opened session.
+// guildID     : Guild ID to create guild-specific application command. If empty - creates global appication command.
+// cmd         : New application command data.
+func (s *Session) ApplicationCommandCreate(appID string, guildID string, cmd *ApplicationCommand) (rcmd *ApplicationCommand, err error) {
+	if appID == "" {
+		appID = s.State.User.ID
+	}
+
+	endpoint := EndpointApplicationGlobalCommands(appID)
+	if guildID != "" {
+		endpoint = EndpointApplicationGuildCommands(appID, guildID)
+	}
+
+	var body []byte
+	body, err = s.RequestWithBucketID("POST", endpoint, *cmd, endpoint)
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &rcmd)
+
+	return
```

Comment:
```return rcmd, nil```

Reply:
No. It's pointless, especially in this function. First of all, `err` isn't always `nil` as in your code which you suggest to change to. And the second point is - I'm using named return values, why I can't use just `return`?

Reply:
Naked retruns hurt readability. As mentioned in the ealier comment by @Strum355 and @eaglemoor it would be preferred to use explicit return values.

Reply:
Then what about usage of naked `return` as whole. As you're saying we don't actually need it, let's just use explicit returns here. Also, why explicit `return` here is better, if I'm clearly using both of them and removing unnecessary code? 
Also, as coding conventions, this is not a just simple and short function tho.

Reply:
> Then what about usage of naked return as whole. As you're saying we don't actually need it, let's just use explicit returns everywhere. 

Please dont strawman argument, its not nice. I've already told you when they are applicable. I have less issues with them in this function (its relatively small) but others would prefer they werent used in general. So yes, some say we should use explicit returns everywhere.

Reply:
Naked returns also make it easy to introduce bugs, see this very example in the PR (which got patched in the last commit). https://github.com/bwmarrin/discordgo/pull/856#discussion_r577859870. 

The last comment on the PR does a great job at explaining my current feelings regarding this PR: https://github.com/bwmarrin/discordgo/pull/856#issuecomment-780835223.

Reply:
> Naked returns also make it easy to introduce bugs, see this very example in the PR (which got patched in the last commit). #856 (comment).
Okay, I think I got your point, you win, I will change this in the code.

Reply:
Thx a lot

Reply:
Just wanted to say, I replaces unnecessary naked returns with normal, explicit returns. But not in all places tho.